src/App.js

```diff
@@ -11,35 +11,115 @@ class App extends React.Component {
+      // nested ternary operator to nest two if statements
+      player1NumRoundsWon:
+        newRoundWinner === 1
+          ? state.player1NumRoundsWon + 1
+          : state.player1NumRoundsWon,
+      player2NumRoundsWon:
+        newRoundWinner === 2
+          ? state.player2NumRoundsWon + 1
+          : state.player2NumRoundsWon,
+    }));
+  };
 
-    // You can access your current components state here, as indicated below
+  render() {
     const currCardElems = this.state.currCards.map(({ name, suit }) => (
       // Give each list element a unique key
       <div key={`${name}${suit}`}>
         {name} of {suit}
       </div>
     ));
 
+    const roundWinnerMessage = this.state.roundWinner
+      ? `Player ${this.state.roundWinner} won this round.`
+      : `This rounds is a tie!`;
+    //Placeholder text when player 1 wins a round
+    const player1RoundsWonMessage = `Player 1 has won ${this.state.player1NumRoundsWon} rounds this game.`;
+    //Placeholder text when player 2 wins a round
+    const player2RoundsWonMessage = `Player 2 has won ${this.state.player2NumRoundsWon} rounds this game.`;
+    // numRoundsLeft is the number of rounds left in the game - divide by two because two cards are drawn each round
+    const numRoundsLeft = this.state.cardDeck.length / 2;
+    const numRoundsLeftMessage = `There are ${numRoundsLeft} rounds left in this game!`;
+
+    // Determine game winner
+    let gameWinner = null;
+    if (this.state.player1NumRoundsWon > this.state.player2NumRoundsWon) {
+      gameWinner = 1;
+    } else if (
+      this.state.player2NumRoundsWon > this.state.player1NumRoundsWon
+    ) {
+      gameWinner = 2;
+    }
+
+    let gameWinnerMessage;
+    if (gameWinner) {
+      gameWinnerMessage = `Player ${gameWinner} won this game!`;
+    } else {
+      gameWinnerMessage = "It's a draw!";
+    }
+
+    // Deal button text changes at end of game to start again
+    let dealButtonText;
+    if (numRoundsLeft === 0) {
+      dealButtonText = "Reset Game";
+    } else {
+      dealButtonText = "Deal";
+    }
+
     return (
       <div className="App">
         <header className="App-header">
           <h3>High Card 🚀</h3>
           {currCardElems}
           <br />
-          <button onClick={this.dealCards}>Deal</button>
+          {/* Button changes functionality depending on game state. When number of rounds = 0, button prompts game reset, else deal cards. */}
+          <button
+            onClick={numRoundsLeft === 0 ? this.resetGame : this.dealCards}
+          >
+            {dealButtonText}
+          </button>
+          <br />
+          <br />
+          {/* Render round winner message if the game has started */}
+          {this.state.hasGameStarted && <p>{roundWinnerMessage}</p>}
+          {this.state.hasGameStarted && <p>{player1RoundsWonMessage}</p>}
+          {this.state.hasGameStarted && <p>{player2RoundsWonMessage}</p>}
+          {this.state.hasGameStarted && <p>{numRoundsLeftMessage}</p>}
```

Comment:
```suggestion
          {this.state.hasGameStarted && (
             <>
                 <p>{roundWinnerMessage}</p>
                 <p>{player1RoundsWonMessage}</p>
                 <p>{player2RoundsWonMessage}</p>
                 <p>{numRoundsLeftMessage}</p>
             </>
          )
```

We don't need to access the same state property multiple times. It won't change value :)